apps/spruce/src/pages/task/taskTabs/TaskHistory/CommitDetailsCard/CommitDescription.tsx

```diff
@@ -0,0 +1,77 @@
+import { useState } from "react";
+import styled from "@emotion/styled";
+import { palette } from "@leafygreen-ui/palette";
+import { Disclaimer } from "@leafygreen-ui/typography";
+import { WordBreak } from "@evg-ui/lib/components/styles";
+import { size } from "@evg-ui/lib/constants/tokens";
+import { useTaskHistoryAnalytics } from "analytics";
+import { useSpruceConfig } from "hooks";
+import { jiraLinkify } from "utils/string";
+
+const { blue } = palette;
+
+interface CommitDetailsCardProps {
+  author: string;
+  message: string;
+}
+
+const CommitDescription: React.FC<CommitDetailsCardProps> = ({
+  author,
+  message,
+}) => {
+  const { sendEvent } = useTaskHistoryAnalytics();
+
+  const spruceConfig = useSpruceConfig();
+  const jiraHost = spruceConfig?.jira?.host ?? "";
+
+  const shouldTruncate = message.length > MAX_CHAR;
+  const truncatedText = message.substring(0, MAX_CHAR).concat("…");
+
+  const [showDescription, setShowDescription] = useState(!shouldTruncate);
+
+  return (
+    <BottomLabel>
+      <AuthorLabel>{author} - </AuthorLabel>
+      <WordBreak>
+        {jiraLinkify(showDescription ? message : truncatedText, jiraHost)}
+        {shouldTruncate ? (
+          <ButtonText
+            onClick={(e) => {
+              e.stopPropagation();
+              setShowDescription(!showDescription);
+              sendEvent({
+                name: "Toggled commit description",
+                expanded: showDescription,
+              });
+            }}
+          >
+            {showDescription ? "Show less" : "Show more"}
+          </ButtonText>
+        ) : null}
+      </WordBreak>
```

Comment:
I think you can shorten this a bit and memoize the linkificaiton to get some minor perf wins. 
```suggestion
  const shouldTruncate = message.length > MAX_CHAR;
  const [showDescription, setShowDescription] = useState(!shouldTruncate);

  const displayMessage = showDescription ? message : `${message.slice(0, MAX_CHAR)}…`;
  const toggleText = showDescription ? "Show less" : "Show more";

  const linkedMessage = useMemo(
    () => jiraLinkify(displayMessage, jiraHost),
    [displayMessage, jiraHost]
  );

  const handleToggle = (e: React.MouseEvent) => {
    e.stopPropagation();
    setShowDescription((prev) => {
      sendEvent({
        name: "Toggled commit description",
        expanded: !prev,
      });
      return !prev;
    });
  };

  return (
    <BottomLabel>
      <AuthorLabel>{author} - </AuthorLabel>
      <WordBreak>
        {linkedMessage}
        {shouldTruncate && (
          <ToggleButton onClick={handleToggle}>{toggleText}</ToggleButton>
        )}
      </WordBreak>
```